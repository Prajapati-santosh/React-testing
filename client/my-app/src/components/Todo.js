import React from "react";

function Todo(){
    return <div>
        <input data-testid="test-1"/>  
        <h1 data-testid="test-2">Hello world</h1>
    </div>;
}

export default Todo;