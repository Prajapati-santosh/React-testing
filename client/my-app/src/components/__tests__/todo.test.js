import { cleanup, render, screen } from '@testing-library/react';
import Todo from '../Todo';
import { Simulate } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';

afterEach(()=>{
    cleanup;
})

test('tests todo component',()=>{
    render(<Todo/>);
    const todoEle=screen.getByTestId('test-2');
    expect(todoEle).toBeInTheDocument();
    expect(todoEle).toHaveTextContent('Hello world');
    

})

test('test todo comopnent button',()=>{
    render(<Todo/>);
    const todoEle=screen.getByTestId('test-1');
    Simulate.input(todoEle,"Hello")
})

test('test to do snapshots',()=>{
    const tree=renderer.create(<Todo/>);
    expect(tree).toMatchSnapshot();
})