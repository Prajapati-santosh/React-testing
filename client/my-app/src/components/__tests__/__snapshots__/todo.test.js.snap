// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to do snapshots 1`] = `
<div>
  <input
    data-testid="test-1"
  />
  <h1
    data-testid="test-2"
  >
    Hello world
  </h1>
</div>
`;
